core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -3323,40 +3395,110 @@ lCompactGeAs ps minElems (Fold step1 initial1 _ final1) =
 
 -- | Like 'compactGE' but for transforming folds instead of stream.
 --
--- >>> lCompactGE n = Fold.many (MutArray.fCompactGE n)
+-- >> lCompactGE n = Fold.many (MutArray.fCompactGE n)
 --
 -- Generates unpinned arrays irrespective of the pinning status of input
 -- arrays.
+{-# DEPRECATED lCompactGE "Please use scanCompactMin instead." #-}
 {-# INLINE lCompactGE #-}
 lCompactGE :: forall m a. (MonadIO m, Unbox a)
     => Int -> Fold m (MutArray a) () -> Fold m (MutArray a) ()
 lCompactGE = lCompactGeAs Unpinned
 
 -- | Pinned version of 'lCompactGE'.
+{-# DEPRECATED lPinnedCompactGE "Please use scanCompactMin' instead." #-}
 {-# INLINE lPinnedCompactGE #-}
 lPinnedCompactGE :: forall m a. (MonadIO m, Unbox a)
     => Int -> Fold m (MutArray a) () -> Fold m (MutArray a) ()
 lPinnedCompactGE = lCompactGeAs Pinned
 
--- | @compactGE n stream@ coalesces adjacent arrays in the @stream@ until
--- the size becomes greater than or equal to @n@.
+data CompactMinState arr =
+    CompactMinInit | CompactMinIncomplete arr | CompactMinComplete arr
+
+{-# INLINE_NORMAL scanCompactMinAs #-}
+scanCompactMinAs :: forall m a. (MonadIO m, Unbox a)
+    => PinnedState -> Int -> Scanl m (MutArray a) (Maybe (MutArray a))
```

Comment:
Instead of `Maybe (MutArray a)`, Why not have `Either (MutArray a) (MutArray a)`?
The use case being: we might have some error handling such as adding padding if `Left`.
Using this, is there a way to handle that?

Reply:
The above statements apply to the last element.
Or the last accumulated chunk